update VTable guide (#755)
vtexdocs/dev-portal-content: docs/guides/Master-Data/master-data-how-it-works/vtable.md

createdAt: "2022-07-28T20:39:04.566Z"
updatedAt: "2022-07-28T21:27:21.269Z"
---

>  VTable is a flexible and dynamic user interface designed to handle data from Master Data v2 directly from VTEX Admin. It allows you to create custom applications for managing and visualizing data efficiently, using [JSON schemas](https://developers.vtex.com/vtex-rest-api/docs/starting-to-work-on-master-data-with-json-schema).

Reviewer: 💭 _obs: pensar em como evitar adjetivos em docs técnicas_

createdAt: "2022-07-28T20:39:04.566Z"
updatedAt: "2022-07-28T21:27:21.269Z"
---

 VTable is a dynamic user interface designed to handle data from Master Data v2 directly from VTEX Admin. It allows you to create custom applications for managing and visualizing data efficiently, using [JSON schemas](https://developers.vtex.com/vtex-rest-api/docs/starting-to-work-on-master-data-with-json-schema).

To render an app in VTable you must follow these steps:

> - [Step by step](#step-by-step)
>   - [Step 1 - Creating a JSON schema](#step-1---creating-a-json-schema)
>   - [Step 2 - Building the app schema](#step-2---building-the-app-schema)
>   - [Step 3 - Validating the app schema](#step-3---validating-the-app-schema)
>   - [Step 4 - Saving the app schema to Master Data](#step-4---saving-the-app-schema-to-master-data)
> - [JSON schema configurations](#json-schema-configurations)

Reviewer: ```suggestion
- [Step 1 - Creating a JSON schema](#step-1---creating-a-json-schema)
- [Step 2 - Building the app schema](#step-2---building-the-app-schema)
- [Step 3 - Validating the app schema](#step-3---validating-the-app-schema)
- [Step 4 - Saving the app schema to Master Data](#step-4---saving-the-app-schema-to-master-data)
```

         "model":"user-v1",
         "fields":{
            "id":{
               "width":200
            },
            "email":{
               "width":200
            },
            "firstName":{
               "width":300
            }
         },
         "list":[
            "email",
            "firstName",
            "lastName"
         ],
         "editor":{
            "settings":{
               "sections":[
                  {
                     "name":"Personal Data",
                     "fields":[
                        "firstName",
                        "email",
                        "lastName"
                     ]
                  }
               ]
            }
         }
      }
   ]
}
```

You can use this example app schema as a foundation to create your own app. Check the meaning of fields in the example app schema:

| Field | Type | Description |
|---|---|---|
| `name` | string | Name of the schema in Master Data v2. |
| `title` | string | App title. |
| `tables` | array of objects | List of objects containing information about the tables. |
| `tables[i].id` | string | Table ID. |
| `tables[i].title` | string | Table title. |

> | `tables[i].entity` | string | Corresponds to the data entity in Master Data, which must exist before creating the app. |

Reviewer: Sugestões apenas para manter o paralelismo da tabela

| `tables[i].model` | string |  Relates to the schema associated with the data entity in Master Data. |
| `tables[i].fields` | object | Lists the fields to be displayed. Each field specifies the width it occupies in the table. |
| `tables[i].fields.id` | object | Field name. |
| `tables[i].fields.id.width` | integer | Width of the `id` field. |
| `tables[i].fields.email` | object | Field name. |
| `tables[i].fields.email.width` | integer | Width of the `email` field. |
|`tables[i].fields.firstName` | object | Field name. |
| `tables[i].fields.firstName.width` | integer | Width of the `firstName` field. |
| `tables[i].list` | array of strings | Defines the fields to be rendered in the table. |

> | `tables[i].editor` | object | Configures the form's rendering. |

Reviewer: ```suggestion
| `tables[i].editor` | object | Object responsible for configuring the form's rendering. |
```

| `tables[i].editor.settings` | object | Form settings. |
| `tables[i].editor.settings.sections` | array of objects | List of form sections and their respective settings. |
| `tables[i].editor.settings.sections[j].name` | string | Setting name. |
| `tables[i].editor.settings.sections[j].fields` | array of strings | List of fields shown in the form. |

Read the [JSON schema configurations](#json-schema-configurations) section to learn how to configure each field for your specific needs.

## 3. Validating the app schema

To validate your app schema, use a tool such as [JSON Schema Validator](https://www.jsonschemavalidator.net/), which allows you to paste the code from [VTable app objects schema](#vtable-app-objects-schema) and the code from your app schema created in [step 2](#2-building-the-app-schema) to check if the structure of your app schema is valid.

## 4. Saving the app schema to Master Data

> To save the app schema to Master Data, you must use the [Create document with custom ID](https://developers.vtex.com/docs/api-reference/master-data-api-v2#put-/api/dataentities/-dataEntityName-/documents/-id-) endpoint, adding your app schema as the request body and filling the URL with the information below:

Reviewer: ```suggestion
To save the app schema to Master Data, send a `PUT` request to the [Create document with custom ID](https://developers.vtex.com/docs/api-reference/master-data-api-v2#put-/api/dataentities/-dataEntityName-/documents/-id-) endpoint, adding your app schema as the request body and filling the URL with the information below:
```

Reviewer: Sugestão com um verbo mais forte, buscando ser mais direta

`PUT` `https://{accountName}.{environment}.com.br/api/dataentities/vtable/documents/{appName}?_schema=app`

Your app schema will be saved as a document in the VTable data entity.

After that, you will be able to access the app data in VTable at `https://{accountName}.myvtex.com/admin/vtable/`, replacing `{accountName}` with your VTEX account name.

> ## JSON schema configurations

Reviewer: Pode valer a pena usar `details` nessa seção para deixar a doc mais concisa. O que acha?

> To render a checkbox the field must have type **boolean**.

Reviewer: Corrigir todas essas descrições, incluindo a vírgula apropriadamente:

To render a checkbox`,` the field must have type **boolean**.

To render a `DatePicker` `,` the field must have the property `format` with the value `date-time`.

....

Outra sugestão seria reescrever essas descrições, usando ações, exemplo:

```
Set the value of the `type` field as `boolean` to render a checkbox.
```
